src/core/tsi/s2a/record_protocol/s2a_crypter.h

```diff
@@ -289,4 +291,57 @@ S2ADecryptStatus s2a_decrypt_record(s2a_crypter* crypter, iovec& record_header,
                                     size_t* plaintext_bytes_written,
                                     char** error_details);
 
+/** This function writes a TLS 1.3 record to |protected_slices| of type
+ *  |record_type|, and with a payload containing the ciphertext obtained by
+ *  encrypting |unprotected_slices| using the outgoing_aead_crypter of
+ *  |crypter|. The arguments of the encrypt function are detailed below:
+ *  - crypter: an instance of s2a_crypter, which must have been initialized
+ *    using the s2a_crypter_create method; in particular, the caller must not
+ *    pass in nullptr for this argument.
+ *  - record_type: a single byte that indicates the TLS record type, i.e. one
+ *    of handshake, application data, alert, or change cipher spec.
+ *  - unprotected_slices: a pointer to the grpc_slice_buffer containing the
+ *    plaintext to be encrypted. The caller must ensure that
+ *    |unprotected_slices| is not nullptr, and the total size of the plaintext
+ *    is at most SSL3_RT_MAX_PLAIN_LENGTH = 16384, otherwise the method returns
+ *    GRPC_STATUS_FAILED_PRECONDITION.
+ *  - protected_slices: a pointer to the grpc_slice_buffer that will be
+ *    populated by the method with a TLS 1.3 record. The caller must ensure that
+ *    |protected_slices| is not nullptr.
+ *  - error_details: the error details generated when the execution of the
+ *    function fails; the caller must not pass in nullptr for this argument.
+ *
+ *  On success, the function returns GRPC_STATUS_OK; otherwise, |error_details|
+ *  is populated with an error message, and it must be freed with gpr_free. If
+ *  the function returns the error code GRPC_STATUS_OUT_OF_RANGE, the caller
+ *  must close the connection. **/
+grpc_status_code s2a_protect_record(s2a_crypter* crypter, uint8_t record_type,
+                                    grpc_slice_buffer* unprotected_slices,
+                                    grpc_slice_buffer* protected_slices,
+                                    char** error_details);
```

Comment:
[optional] I think the new style is to use grpc_error, rather than pass grpc_status_code and char** error_details.

Reply:
Thanks for letting me know. I've used the `grpc_status_code` and `char** error_details` extensively throughout the frame protector and record protocol. If it's ok with you, I will send a separate PR after this one that cleans this up throughout. Does that work?